An unhandled exception occurred starting in 2.51.0

**Describe the bug**
This error has occurred starting with the new version of the dev ops extension 2.51.0. 
The last time I had run this extension successfully it was on the previous version. 
I have not made any changes to my configuration so if any are needed to work with the latest version of the extension please let me know, as I have completely missed those if that's the case. 

**Categorization**
- [x] This is not a permissions issue (Seek help at https://github.com/tinglesoftware/dependabot-azure-devops/discussions/1245)
- [x] This is an issue specific to Azure DevOps or this repository and does not belong in [dependabot-core](https://github.com/dependabot/dependabot-core).
  > Specific issues for dependabot are solved faster in the core repository. For example, why a package version is skipped. Trying out this behaviour in the GitHub Hosted version can help you pinpoint where it lies.
- [x] I have linked a public reproduction of the specific issue or none is required because the issue is not specific to me.
  > Please note that you can create a public organization/project and repository to show the issue. This tends to accelerate resolution.

**Repository**
URL: e.g. https://dev.azure.com/tingle/dependabot/_git/repro-684
If credentials are required, send them via mail

**To Reproduce**
Steps to reproduce the behavior:
1. ...
2. ...

**Expected behavior**
The new version of the extension should run successfully, given that I have not made any configuration changes. 

**Logs and screenshots**
```
2025-06-10T12:21:31.8928920Z ##[error][
  {
    "expected": "string",
    "code": "invalid_type",
    "path": [
      "input",
      "job",
      "ignore-conditions",
      0,
      "updated-at"
    ],
    "message": "Invalid input: expected string, received Date"
  },
  {
    "expected": "string",
    "code": "invalid_type",
    "path": [
      "input",
      "job",
      "ignore-conditions",
      1,
      "updated-at"
    ],
    "message": "Invalid input: expected string, received Date"
  }
]
2025-06-10T12:21:31.8943525Z ##[error]An unhandled exception occurred: ZodError: [
  {
    "expected": "string",
    "code": "invalid_type",
    "path": [
      "input",
      "job",
      "ignore-conditions",
      0,
      "updated-at"
    ],
    "message": "Invalid input: expected string, received Date"
  },
  {
    "expected": "string",
    "code": "invalid_type",
    "path": [
      "input",
      "job",
      "ignore-conditions",
      1,
      "updated-at"
    ],
    "message": "Invalid input: expected string, received Date"
  }
]
2025-06-10T12:21:31.8944499Z ZodError: [
2025-06-10T12:21:31.8944639Z   {
2025-06-10T12:21:31.8944775Z     "expected": "string",
2025-06-10T12:21:31.8944927Z     "code": "invalid_type",
2025-06-10T12:21:31.8945064Z     "path": [
2025-06-10T12:21:31.8945189Z       "input",
2025-06-10T12:21:31.8945316Z       "job",
2025-06-10T12:21:31.8945451Z       "ignore-conditions",
2025-06-10T12:21:31.8945574Z       0,
2025-06-10T12:21:31.8945705Z       "updated-at"
2025-06-10T12:21:31.8945830Z     ],
2025-06-10T12:21:31.8945986Z     "message": "Invalid input: expected string, received Date"
2025-06-10T12:21:31.8946140Z   },
2025-06-10T12:21:31.8946257Z   {
2025-06-10T12:21:31.8946385Z     "expected": "string",
2025-06-10T12:21:31.8946556Z     "code": "invalid_type",
2025-06-10T12:21:31.8946693Z     "path": [
2025-06-10T12:21:31.8946891Z       "input",
2025-06-10T12:21:31.8948243Z       "job",
2025-06-10T12:21:31.8948393Z       "ignore-conditions",
2025-06-10T12:21:31.8948527Z       1,
2025-06-10T12:21:31.8948644Z       "updated-at"
2025-06-10T12:21:31.8948768Z     ],
2025-06-10T12:21:31.8948923Z     "message": "Invalid input: expected string, received Date"
2025-06-10T12:21:31.8949074Z   }
2025-06-10T12:21:31.8949191Z ]
2025-06-10T12:21:31.8949420Z     at readJobScenarioOutputFile (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.0/dist/task-v2.js:105384:51)
2025-06-10T12:21:31.8949763Z     at async DependabotCli.update (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.0/dist/task-v2.js:105309:28)
2025-06-10T12:21:31.8950069Z     at async performDependabotUpdatesAsync (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.0/dist/task-v2.js:110060:25)
2025-06-10T12:21:31.8950371Z     at async run (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.51.0/dist/task-v2.js:109940:24)
```

**Extension (please complete the following information):**
 - Host: Azure DevOps
 - Version 5.51.0

**Server (please complete the following information):**
 - Version ubuntu-24.04

**Additional context**
Add any other context about the problem here.

--- a/.changeset/lazy-cougars-travel.md
+++ b/.changeset/lazy-cougars-travel.md
@@ -1,0 +1,5 @@
+---
+'paklo': patch
+---
+
+Coerce parsing of updated-at in ignore-conditions

--- a/packages/paklo/fixtures/jobs/python-pip.yaml
+++ b/packages/paklo/fixtures/jobs/python-pip.yaml
@@ -5,6 +5,11 @@ job:
   allowed-updates:
     - dependency-type: direct
       update-type: all
+  ignore-conditions:
+    - dependency-name: 'anthropic'
+      update-types: ['version-update:semver-major']
+      source: '.github/dependabot.yml'
+      updated-at: 2025-06-10T13:12:56.154Z
   security-updates-only: false
   security-advisories: []
   source:

--- a/packages/paklo/src/dependabot/job.ts
+++ b/packages/paklo/src/dependabot/job.ts
@@ -57,7 +57,7 @@ export const DependabotConditionSchema = z.object({
   'dependency-name': z.string(),
   'source': z.string().nullish(),
   'update-types': z.string().array().nullish(),
-  'updated-at': z.string().nullish(), // TODO: should we instead use a date here?
+  'updated-at': z.coerce.string().nullish(),
   'version-requirement': z.string().nullish(),
 });
 export type DependabotCondition = z.infer<typeof DependabotConditionSchema>;

